File: golang/Sorting Algorithms/Insertion Sort/insertionSort.go
package main

import "fmt"

func main() {
	arr := []int{0, 5, 3, 5, 7, 4, 6, 9, 4, 4, 7}
	fmt.Printf("Array before Selection Sort\t->\t %v\n", arr)

	insertionSort(arr)

	fmt.Printf("Array after Selection Sort\t->\t %v\n", arr)
}

func insertionSort(arr []int) []int {
	var j, temp int
	for i := 1; i < len(arr); i++ {
		j = i
		temp = arr[i]
		for j > 0 && temp < arr[j-1] {
			arr[j] = arr[j-1]
			j = j - 1
		}
		arr[j] = temp
	}
	return arr
}

File: Java/Sorting Algorithms/InsertionSort.java
 
import java.util.Scanner;
 

public class InsertionSort 
{
    
    public static void sort( int arr[] )
    {
        int N = arr.length;
        int i, j, temp;
        for (i = 1; i< N; i++) 
        {
            j = i;
            temp = arr[i];    
            while (j > 0 && temp < arr[j-1])
            {
                arr[j] = arr[j-1];
                j = j-1;
            }
            arr[j] = temp;            
        }        
    }    
  
    public static void main(String[] args) 
    {
        Scanner scan = new Scanner( System.in );        
        System.out.println("Insertion Sort Test\n");
        int n, i;
       
        System.out.println("Enter number of integer elements");
        n = scan.nextInt();
       
        int arr[] = new int[ n ];
       
        System.out.println("\nEnter "+ n +" integer elements");
        for (i = 0; i < n; i++)
            arr[i] = scan.nextInt();
        
        sort(arr);
     
        System.out.println("\nElements after sorting ");        
        for (i = 0; i < n; i++)
            System.out.print(arr[i]+" ");            
        System.out.println();                     
    }    
}
